frontend/src/app/libs/common-components/components/input/input.component.ts

```diff
@@ -0,0 +1,238 @@
+		this.setSyncValidator();
+		this.setAsyncValidator();
+		this.setValidatorsToFormControl();
+		this.switchToDisable();
+		this.errorsFromFormControl();
+		this.getDefaultValue();
+	}
+
+	public ngAfterViewInit(): void {
+		if (Boolean(this.defaultValue)) {
+			setTimeout(() => {
+					this.formControl.setValue(this.defaultValue);
+			});
+		}
+	}
+
+	public ngOnDestroy(): void {
+		this.ngUnsubscribe.next();
+		this.ngUnsubscribe.complete();
+	}
+
+	public switchToDisable(): void {
+		if (Boolean(this.formControl) && this.disabled) {
+			this.formControl.disable();
+		}
+	}
+
+	public onTouched: any = () => undefined;
+
+	public onChange: any = () => undefined;
+
+	public writeValue(value: any): void {
+		this.value = value;
+		this.onChange(value);
+		if (this.formControl && this.formControl.value) {
+			this.onTouched();
+		}
+		this.valueChange.emit(value);
+	}
+
+	public onValueChange(value: any): void {
+		this.onTouched();
+		if (Boolean(this.value) && this.value !== value) {
+			this.value = value;
+			this.onChange(value);
+			this.valueChange.emit(value);
+		}
+	}
+
+	public onValueInput(value: any): void {
+		this.value = value;
+		this.onChange(value);
+	}
+
+	public registerOnChange(fn: any): void {
+		this.onChange = fn;
+	}
+
+	public registerOnTouched(fn: any): void {
+		this.onTouched = fn;
+	}
+
+	public setValidatorsToFormControl(): void {
+		if (this.syncValidators.length > 0) {this.formControl.setValidators(this.syncValidators); }
+		if (this.asyncValidators.length > 0) {this.formControl.setAsyncValidators(this.asyncValidators); }
+	}
+
+	private getDefaultValue(): void {
+		if (this.formControl?.value && !Boolean(this.defaultValue)) {
+			this.defaultValue = this.formControl?.value;
+		}
+	}
+
+	private setSyncValidator(): void {
+		this.syncValidators = [].concat(this.inputValidators, this.changeValidators, this.inlineSyncValidators)
+		.filter((validator: ValidatorFn | null) => validator);
+	}
+	private setAsyncValidator(): void {
+		this.asyncValidators = [].concat(this.inputValidatorsAsync, this.changeValidatorsAsync, this.inlineAsyncValidators)
+		.filter((validator: ValidatorFn | null) => validator);
+	}
+
+	private getInlineSyncValidatorsFromFormControl(): ValidatorFn[] {
+		if (this.formControl?.validator) {
+			return [].concat(this.formControl.validator);
+		}
+		return null;
+	}
+
+	private getInlineAsyncValidatorsFromFormControl(): AsyncValidatorFn[] {
+		if (this.formControl.asyncValidator) {
+			return [].concat(this.formControl.asyncValidator);
+		}
+		return null;
+	}
+
+	private subscribeToChanges(): void {
+		this.formControl.valueChanges.pipe(takeUntil(this.ngUnsubscribe)).subscribe(() => {
+		this.errorsFromFormControl();
+		});
+	}
+
+	private errorsFromFormControl(): void {
+		setTimeout(() => {
+			this.errorsArray = [];
+			const errors: ValidationErrors = this.formControl.errors;
+			this.getCustomErrorsArray(errors);
+		}, 0);
+	}
+
+	private getCustomErrorsArray(errors: ValidationErrors): void {
```

Comment:
lets ensure that error object that returned by validator contains field `message`
like {
   message: 'test error message`
}
and please ensure that only message property used to create error messages at ui